lib/vector2d.dart

```diff
@@ -2,74 +2,77 @@ import 'dart:math' as math;
 import 'dart:ui' as ui;
 import 'package:box2d_flame/box2d.dart' as b2d;
 
-/// An ordered pair representation (point, position, offset).
+/// An ordered pair representation (point, position, offset, vector2).
 ///
 /// It differs from the default implementations provided (math.Point and ui.Offset) as it's mutable.
 /// Also, it offers helpful converters and a some useful methods for manipulation.
 /// It always uses double values to store the coordinates.
-class Position {
+class Vector2d {
   /// Coordinates
   double x, y;
 
   /// Basic constructor
-  Position(this.x, this.y);
+  Vector2d(this.x, this.y);
 
-  /// Creates a point at the origin
-  Position.empty() : this(0.0, 0.0);
+  /// Creates a [Vector2d] with 0 splat into both lanes of the vector
+  Vector2d.zero() : this(0.0, 0.0);
 
-  /// Creates converting integers to double.
+  /// Creates a [Vector2d] by converting integers to double.
   ///
   /// Internal representation is still using double, the conversion is made in the constructor only.
-  Position.fromInts(int x, int y) : this(x.toDouble(), y.toDouble());
+  Vector2d.fromInts(int x, int y) : this(x.toDouble(), y.toDouble());
 
-  /// Creates using an [ui.Offset]
-  Position.fromOffset(ui.Offset offset) : this(offset.dx, offset.dy);
+  /// Creates a [Vector2d] using an [ui.Offset]
+  Vector2d.fromOffset(ui.Offset offset) : this(offset.dx, offset.dy);
 
-  /// Creates using an [ui.Size]
-  Position.fromSize(ui.Size size) : this(size.width, size.height);
+  /// Creates a [Vector2d] using an [ui.Size]
+  Vector2d.fromSize(ui.Size size) : this(size.width, size.height);
 
-  /// Creates using an [math.Point]
-  Position.fromPoint(math.Point point) : this(point.x, point.y);
+  /// Creates a [Vector2d] using an [math.Point]
+  Vector2d.fromPoint(math.Point point) : this(point.x, point.y);
 
-  /// Creates using another [Position]; i.e., clones this position.
+  /// Clones a [Vector2d]
   ///
   /// This is useful because this class is mutable, so beware of mutability issues.
-  Position.fromPosition(Position position) : this(position.x, position.y);
+  Vector2d.fromVector2d(Vector2d position) : this(position.x, position.y);
 
-  /// Creates using a [b2d.Vector2]
-  Position.fromVector(b2d.Vector2 vector) : this(vector.x, vector.y);
+  /// Creates a [Vector2d] using a [b2d.Vector2]
+  Vector2d.fromVector2(b2d.Vector2 vector) : this(vector.x, vector.y);
 
   /// Adds the coordinates from another vector.
-  ///
-  /// This method changes `this` instance and returns itself.
-  Position add(Position other) {
+  void add(Vector2d other) {
     x += other.x;
     y += other.y;
-    return this;
   }
 
-  /// Substracts the coordinates from another vector.
-  ///
-  /// This method changes `this` instance and returns itself.
-  Position minus(Position other) {
-    return add(other.clone().opposite());
+  /// Subtracts the coordinates from another vector.
+  void minus(Vector2d other) {
```

Comment:
I guess the `return this` pattern is useless in dart because of the `..` operator. Didn't think it through when implementing, this makes it more clear it's a mutator.